dotnet/src/SemanticKernel/Planning/Plan.cs

```diff
@@ -255,9 +252,7 @@ public async Task<Plan> InvokeNextStepAsync(SKContext context, CancellationToken
             var functionVariables = this.GetNextStepVariables(context.Variables, step);
 
             // Execute the step
-            var functionContext = new SKContext(context.Kernel, functionVariables, context.Functions);
-
-            var result = await step.InvokeAsync(functionContext, cancellationToken: cancellationToken).ConfigureAwait(false);
+            var result = await context.Kernel.RunAsync(functionVariables, step, cancellationToken).ConfigureAwait(false);
```

Comment:
If we pass in `kernel` this will become

`var result = await kernel.RunAsync(functionVariables, step, cancellationToken).ConfigureAwait(false);`